desc/protoparse/options.go

```diff
@@ -1002,6 +1002,39 @@ func interpretOptions(res *parseResult, element descriptorish, opts proto.Messag
 	return nil, nil
 }
 
+func loadMessageDescriptorForOptions(l *linker, fd fileDescriptorish, opts proto.Message) (*desc.MessageDescriptor, error) {
+	// see if the file imports a custom version of descriptor.proto
+	fqn := proto.MessageName(opts)
+	d := findMessageDescriptorForOptions(l, fd, fqn)
+	if d != nil {
+		return d, nil
+	}
+	// fall back to built-in options descriptors
+	return desc.LoadMessageDescriptorForMessage(opts)
+}
+
+func findMessageDescriptorForOptions(l *linker, fd fileDescriptorish, messageName string) *desc.MessageDescriptor {
+	d := fd.FindSymbol(messageName)
+	if d != nil {
+		md, _ := d.(*desc.MessageDescriptor)
+		return md
+	}
+
+	// TODO: should this support public imports and be recursive?
```

Comment:
just making sure nothing to do here

Reply:
I think this will involve a little more toying around with protoc to understand any nuances of its behavior and then mimic it exactly.

Also, FWIW, this is only for addressing #360 (using custom `descriptor.proto`), not for implementing the warning on unused imports. I addressed that issue at the same time just since my previous PR happened to already have test cases that cover the custom `descriptor.proto` case. This is a very rarely used feature of protoc (only useful if you are working on protoc or forking it and making changes to `descriptor.proto`). So this PR will at least get protoparse closer even if not yet perfect; so I don't think we need to do anything here right now.